### ISSUE TITLE
Can we have a synchronous version?

### ISSUE THREAD

**Issue by mhkhung** [itinance/react-native-sha256#5]
Handy to use inside a redux reducer..

And may be SHA-512 as well as HMAC-SHA-256/512 as well that are often needed?

**Comment by fungilation** [comment]
Second that.

Also, SHA3 variants like from https://github.com/Caligatio/jsSHA?

**Comment by dluksza** [comment]
+1 for synchronous version

**Comment by lll000111** [comment]
It is not possible to create a synchronous version!

In React Native, *everything* that has to cross the bridge between Javascript and a native module is asynchronous! Either callbacks or promises.

If you want a synchronous version use a pure Javascript SHA-256 package.

Note that [browser crypto (in crypto.subtle.digest)](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest) also returns a promise! Doing crypto is CPU intensive, and anything in Javascript that is CPU bound is better off being done in a separate thread (worker, if it's Javascript). It is BAD DESIGN to build a Javascript application on modern multithreaded n-CPU hardware that starts a CPU intensive task in the main Javascript thread! So you should embrace the asynchronous API for anything crypto.

**Comment by fungilation** [comment]
Agreed. But adding other sha3 variants which was the main issue with this issue (hah) would be nice.

**Comment by lll000111** [comment]
If you don't want to wait, and I made the experience that if you see a package is only "loosely maintained" that it is best not to expect much (there's someone just like you on the other side,, with stuff to do other than implementing stuff for others), you can check out

https://github.com/itinance/react-native-fs/blob/master/android/src/main/java/com/rnfs/RNFSManager.java#L250

and the respective ios file to see how easy it is to add a few lines. Just fork the repo, implement it, then send a pull request :-)

I just to hate this advice when I was on the receiving end, but I think it's one thing when you get it from a busy big repo with many maintainers and one where doing a fork is hard because they keep develping it faster than you can maintain your fork, compared to packages like the one right here, where you are probably better off maintaining your own fork unless the original version is enough.
